huxley/www/js/components/ChairAttendanceView.js

```diff
@@ -67,48 +99,107 @@ var ChairAttendanceView = React.createClass({
+                checked={this.state.country_assignments[country][0].friday_attendance}
+                onChange={this._handleAttendanceChange.bind(this, "friday_attendance", country)}
+              />
+            </label>
+          </td>
+          <td>
+            <label name="session">
+              <input
+                className="choice"
+                type="checkbox"
+                name="Saturday Morning Attendance"
+                checked={this.state.country_assignments[country][0].saturday_morning_attendance}
+                onChange={this._handleAttendanceChange.bind(this, "saturday_morning_attendance", country)}
+              />
+            </label>
+          </td>
+          <td>
+            <label name="session">
+              <input
+                className="choice"
+                type="checkbox"
+                name="Saturday Afternoon Attendance"
+                checked={this.state.country_assignments[country][0].saturday_afternoon_attendance}
+                onChange={this._handleAttendanceChange.bind(this, "saturday_afternoon_attendance", country)}
+              />
+            </label>
+          </td>
+          <td>
+            <label name="session">
+              <input
+                className="choice"
+                type="checkbox"
+                name="Sunday Attendance"
+                checked={this.state.country_assignments[country][0].sunday_attendance}
+                onChange={this._handleAttendanceChange.bind(this, "sunday_attendance", country)}
+              />
+            </label>
+          </td>
+        </tr>
+      );
+    });
+  },
+
+  _handleGetAssignments() {
+    var user = CurrentUserStore.getCurrentUser();
+    var country_assignments = this.state.country_assignments;
+    var delegates = DelegateStore.getCommitteeDelegates(user.committee);
+
+    AssignmentStore.getCommitteeAssignments(user.committee, function(assignments) {
+        for (var delegate of delegates) {
+          var assignment = assignments.find(assignment => assignment.id == delegate.assignment)
+          var countryID = assignment.country;
+          if (countryID in country_assignments) {
+            country_assignments[countryID].push(delegate)
+          } else {
+            country_assignments[countryID] = [delegate]
+          }
+        }
+        this.setState({
+          country_assignments: country_assignments,
+          assignments: assignments,
+        });
+      }.bind(this));
+  },
+
+  _handleAttendanceChange(session, country, event) {
+    var country_assignments = this.state.country_assignments;
+    var delegates = country_assignments[country];
+
+    for (var delegate of delegates) {
+      delegate[session] = !delegate[session];
+    }
+    country_assignments[country] = delegates;
+    this.setState({
+      country_assignments: country_assignments
+    });
+  }, 
+
+  _handleSaveAttendance(event) {
+    this.setState({loading: true});
```

Comment:
Since we're updating optimistically, I don't think we need to show the loading wheel